qcodes/logger/logger.py

```diff
@@ -184,6 +184,63 @@ def flush_telemetry_traces() -> None:
         telemetry_handler.flush()
 
 
+def _create_telemetry_handler() -> AzureLogHandler:
+    """
+    Configure, create, and return the telemetry handler
+    """
+    global telemetry_handler
+
+    # The default_custom_dimensions will appear in the "customDimensions"
+    # field in Azure log analytics for every log message alongside any
+    # custom dimensions that message may have. All messages additionally come
+    # with custom dimensions fileName, level, lineNumber, module, and process
+    default_custom_dimensions = {"pythonExecutable": sys.executable}
+
+    class CustomDimensionsFilter(logging.Filter):
+        """
+        Add application-wide properties to the customDimension field of
+        AzureLogHandler records
+        """
+
+        def __init__(self, custom_dimensions: Dict[str, str]):
+            super().__init__()
+            self.custom_dimensions = custom_dimensions
+
+        def filter(self, record: logging.LogRecord) -> bool:
+            """
+            Add the default custom_dimensions into the current log record
+            """
+            cdim = self.custom_dimensions.copy()
+            cdim.update(getattr(record, "custom_dimensions", {}))
+            record.custom_dimensions = cdim  # type: ignore
```

Comment:
We normally are explicit about which error to ignore. Here I think you should do `# type: ignore[attr-defined]`

Reply:
That's much nicer, thanks.